### PR TITLE
Support custom grep flags via env var (#70)

### PR DESCRIPTION
**Pull request by adsr**
Hacktoberfest entry

### CODE REVIEW

**Comment by bubalush** [comment]
@roopakv Could you check it? Seems that it is a more elegant solution: default is `--i -E`.

File: src/commands/fail_if_dirty.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ steps:
 
         FILES_MODIFIED=""
         setcommit () {
-          FILES_MODIFIED=$(git status -s | grep -i <<# parameters.invert-pattern >> -v <</ parameters.invert-pattern >> -E '<< parameters.pattern >>')
+          FILES_MODIFIED=$(git status -s | grep <<# parameters.invert-pattern >> -v <</ parameters.invert-pattern >> ${SWISSKNIFE_GREP_FLAGS:--i -E} '<< parameters.pattern >>')
```
Comment (roopakv) [inline src/commands/fail_if_dirty.yml:44]: we want the `-i` here irrespective  of invert right?

**Review by roopakv — COMMENTED** [review]
one comment, otherwise LG.

could we maybe move the code to look like so that it is easier to read?
```
FINAL_GREP_FLAGS=${SWISSKNIFE_GREP_FLAGS:--i -E}
...
...
grep $FINAL_GREP_FLAGS ...
```
```

File: src/commands/fail_if_dirty.yml (hunk 1 above)
Comment (adsr) [reply src/commands/fail_if_dirty.yml:44]: The `-i` is part of the default `SWISSKNIFE_GREP_FLAGS` value. So it should expand to `-i -E <patt>` or `-v -i -E <patt>` if invert-pattern is specified. Setting `SWISSKNIFE_GREP_FLAGS` will override `-i -E` in both cases. Is that correct?

**Comment by adsr** [comment]
> one comment, otherwise LG.
> 
> could we maybe move the code to look like so that it is easier to read?
> 
> ```
> FINAL_GREP_FLAGS=${SWISSKNIFE_GREP_FLAGS:--i -E}
> ...
> ...
> grep $FINAL_GREP_FLAGS ...
> ```

Added in f06e248

**Comment by adsr** [comment]
Hi @roopakv. Can you add `hacktoberfest-accepted` label to this issue so that it counts toward my Hacktoberfest goal?

**Review by roopakv — APPROVED** [review]

**Comment by roopakv** [comment]
thank you for the work on this PR @adsr! Sorry it took me a while to get to it. This has been published in version v0.54.0

and I also added the hacktorberfest label for you :)

**Pull request merged**